Dependencies/xr/Include/XR.h

```diff
@@ -120,6 +112,61 @@ namespace xr
         Identifier ID{};
     };
 
+    struct FieldOfView
+    {
+        float AngleLeft;
+        float AngleRight;
+        float AngleUp;
+        float AngleDown;
+    };
+
+    enum class DetectionBoundaryType
+    {
+        Box,
+        Frustum,
+        Sphere
+    };
+
+    struct Frustum
+    {
+        Pose Pose{};
+        FieldOfView FOV{};
+        float FarDistance;
+    };
+
+    struct DetectionBoundary
+    {
+        DetectionBoundaryType Type{ DetectionBoundaryType::Sphere };
+        std::variant<float, Frustum, Vector3f> Data{ 5.f };
+    };
+
+    struct GeometryDetectorOptions
+    {
+        xr::DetectionBoundary DetectionBoundary{};
+        double UpdateInterval{10};
```

Comment:
nit: Do we need extra precision here?

Reply:
iirc, javascript defaults to doubles for number values and this value comes from typescript/javascript. I don't know that anyone will specify this amount of precision, but its currently possible and we can support it.